lua/vex_library/modules/server/e2helperfuncs.lua

```diff
@@ -16,19 +16,89 @@ vex.registerExtension = function(name,enabled,helpstr,...)
     E2Lib.RegisterExtension(name,enabled,helpstr,...)
 end
 
+vex.registerConstant = function(name, value, ...)
+    -- This exists because E2Lib doesn't return a passed value, which can be handy (see selfaware2).
+    E2Lib.registerConstant(string.upper(name), value)
+    return value
+end
+
+vex.newE2Table = function() return {n={},ntypes={},s={},stypes={},size=0} end
+
+local EMPTY_TABLE = vex.newE2Table()
+local istable, type = istable, type
+vex.isE2Table = function(tbl)
+    if not istable(tbl) then return false end
+    --if getmetatable(tbl) then return false end -- E2 table never has metatable attached. (Commented out for now.)
+    for k,v in pairs(tbl) do
+        local def = EMPTY_TABLE[k]
+        if not def then return false end
+        if type(v) ~= type(def) then return false end
+    end
+    return true
+end
+
+local isnumber, isstring, string_lower, table_IsSequential = isnumber, isstring, string.lower, table.IsSequential
+vex.luaTableToE2 = function(tbl,arrayOptimization) -- TODO: Fix cyclic/infinite Lo0oP (see GMod's PrintTable code for reference).
```

Comment:
It is possible for this function to enter an infinite cyclic loop, as the line comment suggests.
[Here's the source code of the `PrintTable`](https://github.com/Facepunch/garrysmod/blob/master/garrysmod/lua/includes/util.lua#L58-L92), see how it keeps the track of *already done tables*...